OpenRA.Mods.Common/Traits/Player/PlayerResources.cs

```diff
@@ -99,11 +123,73 @@ public PlayerResources(Actor self, PlayerResourcesInfo info)
 		[Sync]
 		public int ResourceCapacity;
 
+		readonly Dictionary<string, int> specialResources = new Dictionary<string, int>();
+		public readonly Dictionary<string, int> SpecialResourcesCapacity = new Dictionary<string, int>();
+
 		public int Earned;
 		public int Spent;
 
 		long lastNotificationTime;
 
+		public void GiveSpecialRawResources(int count, string rtype)
```

Comment:
What is `rtype`? Be more verbose.

Reply:
I think milk means resource type?

Reply:
If you don't even know what it means, then it definitely needs a better name!!

Reply:
I will ask milk about this

Reply:
Changed to `resourceValue`